devito/passes/iet/openmp.py

```diff
@@ -135,7 +135,20 @@ def _make_clauses(cls, ncollapse=None, chunk_size=None, nthreads=None,
             clauses.append('num_threads(%s)' % nthreads)
 
         if reduction:
-            args = ','.join(str(i) for i in reduction)
+            reds = []
+            for i in reduction:
+                # OMP expects a size not an index as input of reduction, such as
```

Comment:
this piece of code really looks complicated... Are we totally sure it's really necessary? could you point me to the specs where it states it?

Reply:
Haven't found the exact spec, only went from tutos on openmp 4.5. But 99% sure it's necessary.

Reply:
fine, I guess I can just try running the test and see how it fails without this patch